### PR TITLE
Add js sdk folder to fish-tank

### PR DESCRIPTION
**Pull request by frankfeng98**
This PR added JS SDK folder to fish-tank repo and updated necessary VSCode settings as well as CI check for JS code.

### CODE REVIEW

File: .github/workflows/precommit.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on:
     types: [opened, synchronize, reopened]
 
 jobs:
-  pre-commit:
+  python-pre-commit:
```
Comment (colriot) [inline .github/workflows/precommit.yml:8]: Why do we call this pre-commit if it's already a PR-time check?

File: .github/workflows/precommit.yml [hunk 2]
```diff
@@ -49,3 +49,27 @@ jobs:
...
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+
+      - name: Install dependencies
+        working-directory: ./javascript-sdk
```
Comment (colriot) [inline .github/workflows/precommit.yml:66]: we should just call the folder `js`. it's not SDK, and JS a pretty widespread short name

File: javascript-sdk/examples/first-steps/first-steps.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+async function extractProductData(page, searchKeyWord) {
+  // Find DOM element using AgentQL API's queryElements() method
+  const response = await page.queryElements(SEARCH_BOX_QUERY);
+
+  // Interact with the element using Playwright API
+  await response.search_product_box.type(searchKeyWord, { delay: 200 });
+  await page.keyboard.press('Enter');
+
+  // Extract data using AgentQL API's queryData() method
+  const data = await page.queryData(PRODUCT_DATA_QUERY);
```
Comment (colriot) [inline javascript-sdk/examples/first-steps/first-steps.js:57]: nit: inline `data`

**Review by colriot — APPROVED** [review]
Unblock the dogfooding

**Review by paveldudka — DISMISSED** [review]

File: .github/workflows/precommit.yml
[the remark this answers is not in the document]
Comment (frankfeng98) [reply .github/workflows/precommit.yml:1]: Understood! I will update it in the upcoming PRs. For now, I will leave as it is so people could merge into this branch.

**Review by rachelnabors — APPROVED** [review]

**Comment by frankfeng98** [comment]
TODO:

1. Update all files to be `main.py` or `main.js`
2. Update `wrap()` to be async
3. Add back `configure()` since it does not have clear downside but may provide benefits if users skip README and directly copy-paste code? cc: @rachelnabors @colriot

**Pull request merged**